sim/hunter/marksmanship/marksmanship.go

```diff
@@ -22,20 +22,66 @@ func RegisterMarksmanshipHunter() {
 		},
 	)
 }
+func (hunter *MarksmanshipHunter) applyMastery() {
+	actionID := core.ActionID{SpellID: 76659}
 
+	wqSpell := hunter.RegisterSpell(core.SpellConfig{
+		ActionID:    actionID,
+		SpellSchool: core.SpellSchoolNature,
+		ProcMask:    core.ProcMaskRangedAuto,
+		Flags:       core.SpellFlagNoOnCastComplete,
+
+		DamageMultiplier: 0.8, // Wowwiki says it remains 80%
+		CritMultiplier:   hunter.CritMultiplier(false, false, false),
+		ThreatMultiplier: 1,
+
+		ApplyEffects: func(sim *core.Simulation, target *core.Unit, spell *core.Spell) {
+			baseDamage := spell.Unit.RangedWeaponDamage(sim, spell.RangedAttackPower(target)) +
+				spell.BonusWeaponDamage()
+			spell.CalcAndDealDamage(sim, target, baseDamage, spell.OutcomeRangedHitAndCrit)
+		},
+	})
+
+	hunter.RegisterAura(core.Aura{
+		Label:    "Wild Quiver Mastery",
+		Duration: core.NeverExpires,
+		OnReset: func(aura *core.Aura, sim *core.Simulation) {
+			aura.Activate(sim)
+		},
+		OnSpellHitDealt: func(aura *core.Aura, sim *core.Simulation, spell *core.Spell, result *core.SpellResult) {
+			if spell != hunter.AutoAttacks.RangedAuto() {
+				return
+			}
+
+			procChance := 0.168 + (hunter.CalculateMasteryPoints() * 0.021) // Todo: Is this right scaling?
+			if sim.RandomFloat("Wild Quiver") < procChance {
+				wqSpell.Cast(sim, result.Target)
+			}
+		},
+	})
+}
 func NewMarksmanshipHunter(character *core.Character, options *proto.Player) *MarksmanshipHunter {
 	mmOptions := options.GetMarksmanshipHunter().Options
 
 	mmHunter := &MarksmanshipHunter{
 		Hunter: hunter.NewHunter(character, options, mmOptions.ClassOptions),
 	}
 	mmHunter.MarksmanshipOptions = mmOptions
-
 	return mmHunter
 }
+func (mmHunter *MarksmanshipHunter) Initialize() {
+	mmHunter.Hunter.Initialize()
+
+	mmHunter.aimedShotTimer = mmHunter.NewTimer()
+	mmHunter.registerAimedShotSpell(mmHunter.aimedShotTimer)
```

Comment:
the only reason for passing this timemr would be if you want to pass it to another spell and make them share CD. If its used in a single spell its better to not pass anything and create a new timer in the spell registration

Reply:
If you want to reset the timer of the AimedShot then you can just do `hunter.AimedShot.CD.Reset()` so you dont need to hold a reference to this timer in the MM struct

Reply:
Yeah, I think that was just an inheritance from wotlk. Cata AiS doesnt have a cd, fixed